### PR TITLE
Catch all routes + Dynamic Profiles

### PR DESCRIPTION
**Pull request by kilemensi**
## Description

The content team needs ability to add dynamic profiles to the CMS without requiring TECH team's help.
Nextjs has all introduce [`Catch all routes`](https://nextjs.org/docs/routing/dynamic-routes#catch-all-routes) which changes how dynamic routes works.

This PR upgrades our dynamic routing mechanism as well as enable dynamic profiles that do not need a specific country. 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Screenshots

Screenshot_2020-03-27 COVID-19 Takwimu

ASIDE: Next.js now requires 404 error page be static and in a `404.js` file... `_error.js` remains for other "dynamic" errors.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/pages/embed/[...path].js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+Embed.defaultProps = {
+  error: undefined,
+  id: undefined,
+  type: undefined,
+  geoId: undefined,
+  definition: undefined
+};
+
+Embed.getInitialProps = async ({ query: { path } }) => {
+  console.log('BOOM', { path });
```
Comment (karimkawambwa) [inline src/pages/embed/[...path].js]:
```suggestion
```

File: src/pages/profiles/[...geoIdOrCountrySlug].js [hunk 2]
```diff
@@ -4,21 +4,21 @@ import PropTypes from 'prop-types';
...
+  loader: () => import('../../components/AnalysisPage')
 });
 const Profile = dynamic({
   ssr: true,
-  loader: () => import('../../../components/ProfilePage')
+  loader: () => import('../../components/ProfilePage')
 });
 
-function ProfileOrAnalysisPage({ isAnalysis, initialProps }) {
+function ProfileOrAnalysisPage({ isProfile, initialProps }) {
```
Comment (karimkawambwa) [inline src/pages/profiles/[...geoIdOrCountrySlug].js]: 😆 ?

Comment (kilemensi) [reply src/pages/profiles/[...geoIdOrCountrySlug].js]: I wish... had to reverse from _if not analysis, assume profile_ to _if not profile, assume analysis_ to support dynamic profiles.

File: src/pages/embed/[...path].js (hunk 1 above)
Comment (kilemensi) [reply src/pages/embed/[...path].js]: It's got to go **BOOM**

**Pull request merged**